modules/kms_crypto_keys_iam/main.tf

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Locals configuration for module logic
+ *****************************************/
+locals {
+  authoritative        = var.mode == "authoritative" ? 1 : 0
+  additive             = var.mode == "additive" ? 1 : 0
+  kms_crypto_key_count = var.kms_crypto_keys_num == 0 ? length(var.kms_crypto_keys) : var.kms_crypto_keys_num
```

Comment:
I am not sure this will work with dynamic values later

Reply:
It tested and just works